### PR TITLE
Update Dockerfile for 4.0

### PR DESCRIPTION
**Pull request by emre-aydin**
* Use root context path by default
* Change MC home directory to `/opt/hazelcast/management-center`
* Remove usage of legacy MC home directory command line argument
* Use the new MC artifact name
* Update `Launcher` class name in start script to include its package name

~NOTE: This PR should NOT be merged before 4.0 is released. Before merging this PR, we need to create a `3.12.x` branch for Docker images for 3.12.x line.~

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,7 +36,7 @@ RUN wget -O ${MC_HOME}/${MC_INSTALL_ZIP} \
  && rm -rf ${MC_INSTALL_NAME}
 
 # Runtime environment variables
-ENV JAVA_OPTS_DEFAULT "-Dhazelcast.mancenter.home=${MC_DATA} -Djava.net.preferIPv4Stack=true"
+ENV JAVA_OPTS_DEFAULT "-Dhazelcast.mc.home=${MC_DATA} -Djava.net.preferIPv4Stack=true"
```
Comment (erosb) [inline Dockerfile]: In the MC codebase do we want to maintain / do anything with the `ui-tests` testsuite? It still refers to the old `hazelcast.mancenter.home` sys prop at 3 occurences).

File: README.md [hunk 2]
```diff
@@ -13,17 +13,17 @@ You can launch Hazelcast Management Center by simply running the following comma
...
 ### Management Center Default Context Path
 
-Before version 3.10, default context path was `/mancenter`, so you would access Hazelcast Management 
-Center by using `http://localhost:8080/mancenter`. Starting with version 3.10, it is changed to
-`/hazelcast-mancenter`, so you can access it by using `http://localhost:8080/hazelcast-mancenter`.
+Before version 4.0, default context path was `/hazelcast-mancenter`, so you would access Hazelcast 
+Management Center by using `http://localhost:8080/hazelcast-mancenter`. Starting with version 4.0, 
+it is changed to root context path (i.e. `/`), so you can access it by using `http://localhost:8080`.
 
 You can overwrite this default by setting the environment variable `MC_CONTEXT`.
```
Comment (erosb) [inline README.md]: Again not strictly related to this PR, but can we rename this env var to something more meaningful? Eg. `MC_BASE_PATH` or similar? We could take the 4.0 release as an opportunity to make this breaking change.

File: Dockerfile (hunk 1 above)
Comment (emre-aydin) [reply Dockerfile]: Those will need to be updated when the `ui-tests` are migrated to work on `master`. @yuraku please make sure you use the new parameters in UI tests.

File: README.md (hunk 2 above)
Comment (emre-aydin) [reply README.md]: AFAIK context path is a widely used term. Are you OK to change it as `MC_CONTEXT_PATH`?

Comment (erosb) [reply README.md]: Sure.

File: Dockerfile (hunk 1 above)
Comment (yuraku) [reply Dockerfile]: ui-tests will be removed in scope of https://github.com/hazelcast/management-center/pull/2595

**Pull request merged**